src/app/core/search/search.service.ts

```diff
@@ -37,12 +39,19 @@ export class SearchService {
 
   getSearchInputs(jurisdictionId: string, caseTypeId: string): Observable<SearchInput[]> {
     let url = this.getSearchInputUrl(jurisdictionId, caseTypeId);
+    this.currentJurisdiction = jurisdictionId;
+    this.currentCaseType = caseTypeId;
     return this.httpService
       .get(url)
       .map(response => {
         let searchInputs = response.json();
-        searchInputs
-          .forEach( item => { item.field.label = item.label; });
+        if (!(this.currentJurisdiction !== jurisdictionId || this.currentCaseType !== caseTypeId)) {
```

Comment:
Is it not simpler to write:
this.currentJurisdiction === jurisdictionId && this.currentCaseType === caseTypeId
and maybe wrap it in a method hasRequiredData or the like?